examples/mlperf/lr_schedulers.py

```diff
@@ -0,0 +1,21 @@
+from tinygrad import Tensor
+from tinygrad.nn.optim import Optimizer
+
+from extra.lr_scheduler import LR_Scheduler
+
+# https://github.com/mlcommons/training/blob/e237206991d10449d9675d95606459a3cb6c21ad/image_classification/tensorflow2/lars_util.py
+class PolynomialDecayWithWarmup(LR_Scheduler):
+  def __init__(self, optimizer: Optimizer, initial_lr, end_lr, train_steps, warmup, power=2):
+    super().__init__(optimizer)
+    assert train_steps > 0 and warmup > 0
+    self.warmup = min(warmup, train_steps)
+    self.initial_lr, self.end_lr, self.epochs, self.power = initial_lr, end_lr, train_steps, power
+
+    # set lr for first warmup step
+    self.optimizer.lr.assign(lr.cast(self.optimizer.lr.dtype) if isinstance(lr := self.get_lr(), Tensor) else lr).realize()
```

Comment:
why would `self.get_lr` not be a Tensor? can you annotate output type for `get_lr`?

Reply:
unfortunately lr_schedule.py is really old. half of the schedulers don't work with jit, and some of them return floats. needs work.

Reply:
the reason why this line is so long is because resnet train_steps is larger than the fp16 largest int value, so we need epoch_counter in fp32 always, and thus we need to cast the output of get_lr. thought i would do a separate pr for that, though

Reply:
i think break it into two lines with a TODO note helps, it's fine to fix other scheduler in a separate pr

Reply:
needs other changes in lr_scheduler.py anyways, will defer them until resnet fp16 PR